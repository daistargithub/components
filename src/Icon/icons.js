import check from 'check-types'
import { themeGet } from 'defaultTheme'

const requireSvg = require.context('./svgIcons', true, /\.svg$/)

export const getIcon = key => {
  try {
    return requireSvg(`./${key}.svg`)
  } catch (ex) {
    console.warn(ex)
    return null
  }
}

export const iconKeys = requireSvg
  .keys()
  .map(key => key.replace('./', '').replace('.svg', ''))

const iconPropType = (props, propName) => {
  const value = props[propName]

  if (
    iconKeys.includes(value) ||
    check.string(themeGet(`icons.${value}`)(props))
  ) {
    return null
  }

  return new TypeError(`Invalid icon value: ${value}`)
}

export const propTypes = {
  icon: iconPropType
}
