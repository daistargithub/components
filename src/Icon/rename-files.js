#!/usr/bin/env node
const changeCase = require('change-case')

const fs = require('fs')
const path = require('path')
const dir = path.resolve(__dirname, 'svgIcons')
const files = fs.readdirSync(dir)

files.forEach(file => {
  if (file.indexOf('.svg') !== -1) {
    const filePath = path.join(dir, file).split('/')
    const filename = filePath.pop().split('.')
    const extension = filename.pop()
    const newFilename = changeCase.snakeCase(filename[0])
    const newFilePath = `${filePath.join('/')}/${newFilename}.${extension}`
    fs.renameSync(path.join(dir, file), newFilePath)
  }
})
