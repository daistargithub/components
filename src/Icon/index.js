import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import check from 'check-types'
import { space, style, px } from 'styled-system'
import { getIcon, propTypes as iconPropTypes } from './icons'
import { themeGet } from '../defaultTheme'

/**
 * Renders the icons provided with this library
 *
 * @param icon {string}
 * @param rest
 */
export const DefaultIconStrategy = ({ icon, ...rest }) => {
  const Component = getIcon(icon)

  if (!Component) return null

  return <Svg as={Component} {...rest} />
}

DefaultIconStrategy.propTypes = {
  ...iconPropTypes
}

export const IconContext = React.createContext({
  iconStrategy: DefaultIconStrategy
})

const height = style({
  prop: 'size',
  key: 'iconSizes',
  cssProperty: 'height',
  transformValue: px
})

const width = style({
  prop: 'size',
  key: 'iconSizes',
  cssProperty: 'width',
  transformValue: px
})

const fill = style({
  prop: 'color',
  key: 'colors',
  cssProperty: 'fill'
})

const hover = ({ onClick }) =>
  check.function(onClick) && '&:hover { cursor: pointer; }'

export const Svg = styled.svg`
  ${hover};
  ${width};
  ${height};
  ${fill};
  ${space};
  ${hover};
  & * {
    ${fill};
  }
`

export const Icon = props => {
  return (
    <IconContext.Consumer>
      {({ iconStrategy }) => iconStrategy(props)}
    </IconContext.Consumer>
  )
}

Icon.propTypes = {
  ...fill.propTypes,
  ...height.propTypes,
  ...width.propTypes
}

Icon.defaultProps = {
  color: 'light',
  size: 3,
  icon: null
}
