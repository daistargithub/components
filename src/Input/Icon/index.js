import styled from 'styled-components'
import { Icon } from 'Icon'

export const InputIcon = styled(Icon).attrs({
  size: 3
})`
  flex: 0 0 auto;
  vertical-align: middle;
`
