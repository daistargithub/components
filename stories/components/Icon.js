import React from 'react'
import { storiesOf } from '@storybook/react'
import { Flex } from 'FlexBox'
import { Pane } from 'Pane'
import { Icon, IconContext, DefaultIconStrategy } from 'Icon'
import { Text } from 'Text'
import { iconKeys } from 'Icon/icons'
import MaterialIcon from '@material/react-material-icon'
import styled from 'styled-components'
import { themeGet } from 'defaultTheme'

// eslint-disable-next-line import/no-webpack-loader-syntax
const materialIcons = require('raw-loader!material-design-icons/iconfont/codepoints')
  .split('\n')
  .map(icon => icon.split(' ').shift())

const StyledMaterialIcon = styled(MaterialIcon)`
  color: ${({ color }) => themeGet(`colors.${color}`, color)};
`

const materialIconStrategy = props => {
  const { icon, ...rest } = props

  if (materialIcons.includes(icon)) {
    const fontSize = themeGet(`iconSizes.${rest.size}`)(props) || props.size

    return <StyledMaterialIcon style={{ fontSize }} icon={icon} {...rest} />
  }

  return <DefaultIconStrategy {...props} />
}

const ExampleIcon = props => {
  const { icon, ...rest } = props

  return (
    <Pane
      key={icon}
      minWidth={50}
      mr={3}
      mb={3}
      style={{ textAlign: 'center' }}
    >
      <Icon icon={icon} {...rest} />
      <Text>{icon}</Text>
    </Pane>
  )
}

const AllIcons = props => (
  <Flex justifyContent='center' alignItems='center' flexWrap='wrap'>
    {iconKeys.map(icon => (
      <ExampleIcon {...props} key={icon} icon={icon} />
    ))}
  </Flex>
)

storiesOf('Icon', module)
  .add('Default', () => <AllIcons />)
  .add('With color', () => <AllIcons color='dark' />)
  .add('With theme size', () => <AllIcons size={5} />)
  .add('With custom size', () => <AllIcons size='18px' />)
  .add('With all', () => <AllIcons size={3} color='darker' />)
  .add('With custom icons', () => (
    <React.Fragment>
      <link
        href='https://fonts.googleapis.com/icon?family=Material+Icons'
        rel='stylesheet'
      />
      <Flex justifyContent='center' alignItems='center' flexWrap='wrap'>
        <IconContext.Provider value={{ iconStrategy: materialIconStrategy }}>
          {materialIcons.map(icon => (
            <ExampleIcon key={icon} icon={icon} />
          ))}
        </IconContext.Provider>
      </Flex>
    </React.Fragment>
  ))
  .add('With custom icons and custom size', () => (
    <React.Fragment>
      <link
        href='https://fonts.googleapis.com/icon?family=Material+Icons'
        rel='stylesheet'
      />
      <Flex justifyContent='center' alignItems='center' flexWrap='wrap'>
        <IconContext.Provider value={{ iconStrategy: materialIconStrategy }}>
          {materialIcons.map(icon => (
            <ExampleIcon key={icon} icon={icon} size='44px' color='light' />
          ))}
        </IconContext.Provider>
      </Flex>
    </React.Fragment>
  ))
