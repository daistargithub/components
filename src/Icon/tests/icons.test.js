import React from 'react'
import { containerRenderer as renderer } from 'utils/tests/containerRenderer'
import 'jest-styled-components'
import { Icon } from 'Icon'
import { iconKeys } from 'Icon/icons'

test('Icon keys have not changed', () => {
  expect(iconKeys).toMatchSnapshot()
})

test('Placeholder is shown when no icon is provided', () => {
  const component = renderer.create(<Icon />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Icon width / height is modified by the size property', () => {
  const component = renderer.create(<Icon size='50px' />)

  let tree = component.toJSON()
  expect(tree).toHaveStyleRule('height', '50px')
  expect(tree).toHaveStyleRule('width', '50px')
})

test('Icon fill is modified by the color property', () => {
  const component = renderer.create(<Icon color='red' />)

  let tree = component.toJSON()
  expect(tree).toHaveStyleRule('fill', 'red')
})

iconKeys.forEach(icon => {
  test(`${icon} renders correctly`, () => {
    const component = renderer.create(<Icon icon={`${icon}`} />)

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
