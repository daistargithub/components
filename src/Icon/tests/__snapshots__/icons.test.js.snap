// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon keys have not changed 1`] = `
Array [
  "keyboard_arrow_up",
  "chevron_right",
  "chevron_down",
  "chevron_left",
  "more_vert",
  "search",
  "close",
  "check",
  "placeholder",
  "empty",
  "hashtag",
  "poll",
  "round_checked",
  "checkbox",
  "round_unchecked",
  "uncheckbox",
  "caret_up",
  "arrow-right",
  "caret_down",
  "arrow-left",
  "arrow_up",
  "arrow_down",
  "document",
]
`;

exports[`Placeholder is shown when no icon is provided 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <rect
    height="24"
    rx="2"
    width="24"
    x="0"
    y="0"
  />
</svg>
`;

exports[`caret_down renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <polygon
    points="9.5 17.5 14.5 12.5 9.5 7.5"
    transform="translate(12, 12.5) rotate(90) translate(-12, -12.5)"
  />
</svg>
`;

exports[`arrow_down renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <polygon
    points="2.66666667 8 3.60666667 8.94 7.33333333 5.22 7.33333333 13.3333333 8.66666667 13.3333333 8.66666667 5.22 12.3866667 8.94666667 13.3333333 8 8 2.66666667"
    transform="translate(21, 21.5) rotate(180)"
  />
</svg>
`;

exports[`arrow-left renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <polygon
    points="9.5 17.5 14.5 12.5 9.5 7.5"
    transform="translate(12, 12.5) rotate(180) translate(-12, -12.5)"
  />
</svg>
`;

exports[`arrow-right renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <polygon
    points="9.5 17.5 14.5 12.5 9.5 7.5"
  />
</svg>
`;

exports[`caret_up renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <polygon
    points="9.5 17.5 14.5 12.5 9.5 7.5"
    transform="translate(12, 12.5) rotate(270) translate(-12, -12.5)"
  />
</svg>
`;

exports[`arrow_up renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <polygon
    points="2.66666667 8 3.60666667 8.94 7.33333333 5.22 7.33333333 13.3333333 8.66666667 13.3333333 8.66666667 5.22 12.3866667 8.94666667 13.3333333 8 8 2.66666667"
    transform="translate(5, 5)"
  />
</svg>
`;

exports[`check renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <path
    d="M8.8,15.9049909 L5.3,12.4049909 C5.11514633,12.2178428 4.86304993,12.1125091 4.6,12.1125091 C4.33695007,12.1125091 4.08485367,12.2178428 3.9,12.4049909 C3.51,12.7949909 3.51,13.4149909 3.9,13.8049909 L8.09,17.9949909 C8.48,18.3849909 9.11,18.3849909 9.5,17.9949909 L20.1,7.40499087 C20.49,7.01499087 20.49,6.39499087 20.1,6.00499087 C19.9151463,5.81784283 19.6630499,5.71250913 19.4,5.71250913 C19.1369501,5.71250913 18.8848537,5.81784283 18.7,6.00499087 L8.8,15.9049909 Z"
    id="path-1"
  />
</svg>
`;

exports[`round_checked renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <path
    d="M12,21 C7.0275,21 3,16.9725 3,12 C3,7.0275 7.0275,3 12,3 C16.9725,3 21,7.0275 21,12 C21,16.9725 16.9725,21 12,21 Z M9.29,16.29 C9.68,16.68 10.32,16.68 10.7,16.29 L18.29,8.7 C18.68,8.31 18.68,7.68 18.29,7.29 C17.9,6.9 17.27,6.9 16.88,7.29 L10,14.17 L7.11,11.29 C6.72,10.9 6.09,10.9 5.7,11.29 C5.51274733,11.4768315 5.40751439,11.7304824 5.40751439,11.995 C5.40751439,12.2595176 5.51274733,12.5131685 5.7,12.7 L9.29,16.29 Z"
  />
</svg>
`;

exports[`checkbox renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <path
    d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
  />
</svg>
`;

exports[`close renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <path
    d="M12,0.75 C5.77875,0.75 0.75,5.77875 0.75,12 C0.75,18.22125 5.77875,23.25 12,23.25 C18.22125,23.25 23.25,18.22125 23.25,12 C23.25,5.77875 18.22125,0.75 12,0.75 Z M16.8375,16.8375 C16.39875,17.27625 15.69,17.27625 15.25125,16.8375 L12,13.58625 L8.74875,16.8375 C8.31,17.27625 7.60125,17.27625 7.1625,16.8375 C6.95184073,16.6273146 6.8334537,16.3419573 6.8334537,16.044375 C6.8334537,15.7467927 6.95184073,15.4614354 7.1625,15.25125 L10.41375,12 L7.1625,8.74875 C6.95184073,8.53856454 6.8334537,8.25320734 6.8334537,7.955625 C6.8334537,7.65804265 6.95184073,7.37268546 7.1625,7.1625 C7.60125,6.72375 8.31,6.72375 8.74875,7.1625 L12,10.41375 L15.25125,7.1625 C15.69,6.72375 16.39875,6.72375 16.8375,7.1625 C17.27625,7.60125 17.27625,8.31 16.8375,8.74875 L13.58625,12 L16.8375,15.25125 C17.265,15.67875 17.265,16.39875 16.8375,16.8375 Z"
  />
</svg>
`;

exports[`document renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <path
    d="M15,1 L4,1 C2.9,1 2,1.9 2,3 L2,16 C2,16.55 2.45,17 3,17 C3.55,17 4,16.55 4,16 L4,4 C4,3.45 4.45,3 5,3 L15,3 C15.55,3 16,2.55 16,2 C16,1.45 15.55,1 15,1 Z M15.59,5.59 L20.42,10.42 C20.79,10.79 21,11.3 21,11.83 L21,21 C21,22.1 20.1,23 19,23 L7.99,23 C6.89,23 6,22.1 6,21 L6.01,7 C6.01,5.9 6.9,5 8,5 L14.17,5 C14.7,5 15.21,5.21 15.59,5.59 Z M15,12 L19.5,12 L14,6.5 L14,11 C14,11.55 14.45,12 15,12 Z"
  />
</svg>
`;

exports[`empty renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
/>
`;

exports[`hashtag renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <path
    d="M3,5 C3,3.8954305 3.8954305,3 5,3 L19,3 C20.1045695,3 21,3.8954305 21,5 L21,19 C21,20.1045695 20.1045695,21 19,21 L5,21 C3.89,21 3,20.1 3,19 L3,5 Z M9.07241672,17.5861902 L9.35,16 L13.35,16 L13.1025833,17.4138098 C13.0549818,17.6858185 13.2368997,17.9449137 13.5089083,17.9925152 C13.5373682,17.9974957 13.5662062,18 13.5950985,18 L14.5799015,18 C14.8227913,18 15.0305473,17.825444 15.0724167,17.5861902 L15.35,16 L16.9319646,16 C17.1739397,16 17.3811896,15.8267244 17.4240562,15.5885765 L17.6040562,14.5885765 C17.6529757,14.3168018 17.4723158,14.0568278 17.2005411,14.0079083 C17.1713091,14.0026466 17.1416664,14 17.1119646,14 L15.71,14 L16.41,10 L17.9899015,10 C18.2327913,10 18.4405473,9.82544403 18.4824167,9.58619017 L18.6574167,8.58619017 C18.7050182,8.31418151 18.5231003,8.05508628 18.2510917,8.00748477 C18.2226318,8.00250429 18.1937938,8 18.1649015,8 L16.76,8 L17.0140562,6.5885765 C17.0629757,6.31680178 16.8823158,6.05682779 16.6105411,6.00790834 C16.5813091,6.00264658 16.5516664,6 16.5219646,6 L15.5380354,6 C15.2960603,6 15.0888104,6.17327564 15.0459438,6.4114235 L14.76,8 L10.76,8 L11.0140562,6.5885765 C11.0629757,6.31680178 10.8823158,6.05682779 10.6105411,6.00790834 C10.5813091,6.00264658 10.5516664,6 10.5219646,6 L9.53803543,6 C9.29606033,6 9.08881038,6.17327564 9.04594377,6.4114235 L8.76,8 L7.18009851,8 C6.9372087,8 6.72945271,8.17455597 6.68758328,8.41380983 L6.51258328,9.41380983 C6.46498176,9.68581849 6.64689969,9.94491372 6.91890835,9.99251523 C6.94736819,9.99749571 6.97620617,10 7.00509851,10 L8.41,10 L7.71,14 L6.12803543,14 C5.88606033,14 5.67881038,14.1732756 5.63594377,14.4114235 L5.45594377,15.4114235 C5.40702432,15.6831982 5.58768421,15.9431722 5.85945893,15.9920917 C5.88869093,15.9973534 5.91833365,16 5.94803543,16 L7.35,16 L7.10258328,17.4138098 C7.05498176,17.6858185 7.23689969,17.9449137 7.50890835,17.9925152 C7.53736819,17.9974957 7.56620617,18 7.59509851,18 L8.57990149,18 C8.8227913,18 9.03054729,17.825444 9.07241672,17.5861902 Z M10.41,10 L14.41,10 L13.71,14 L9.71,14 L10.41,10 Z"
  />
</svg>
`;

exports[`chevron_down renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <path
    d="M15.875,8.9999928 L11.995,12.8799928 L8.115,8.9999928 C7.92816847,8.81274013 7.67451764,8.7075072 7.41,8.7075072 C7.14548236,8.7075072 6.89183153,8.81274013 6.705,8.9999928 C6.315,9.3899928 6.315,10.0199928 6.705,10.4099928 L11.295,14.9999928 C11.685,15.3899928 12.315,15.3899928 12.705,14.9999928 L17.295,10.4099928 C17.685,10.0199928 17.685,9.3899928 17.295,8.9999928 C16.905,8.6199928 16.265,8.6099928 15.875,8.9999928 Z"
  />
</svg>
`;

exports[`chevron_left renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <path
    d="M15,6.7a1,1,0,0,0-1.41,0L9,11.29A1,1,0,0,0,9,12.7l4.59,4.59A1,1,0,1,0,15,15.88L11.12,12,15,8.11A1,1,0,0,0,15,6.7Z"
    style={
      Object {
        "transform": "translate(-24px -24px)",
      }
    }
  />
</svg>
`;

exports[`chevron_right renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <path
    d="M9,6.71A1,1,0,0,0,9,8.12L12.88,12,9,15.88a1,1,0,1,0,1.41,1.41L15,12.7a1,1,0,0,0,0-1.41L10.41,6.7A1,1,0,0,0,9,6.71Z"
    style={
      Object {
        "transform": "translate(-24px -24px)",
      }
    }
  />
</svg>
`;

exports[`keyboard_arrow_up renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <path
    d="M11.295,9 L6.705,13.59 C6.315,13.98 6.315,14.61 6.705,15 C7.095,15.39 7.725,15.39 8.115,15 L12.005,11.12 L15.885,15 C16.275,15.39 16.905,15.39 17.295,15 C17.685,14.61 17.685,13.98 17.295,13.59 L12.705,9 C12.325,8.61 11.685,8.61 11.295,9 Z"
  />
</svg>
`;

exports[`more_vert renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <path
    d="M6,8 C7.1,8 8,7.1 8,6 C8,4.9 7.1,4 6,4 C4.9,4 4,4.9 4,6 C4,7.1 4.9,8 6,8 L6,8 Z M6,10 C4.9,10 4,10.9 4,12 C4,13.1 4.9,14 6,14 C7.1,14 8,13.1 8,12 C8,10.9 7.1,10 6,10 L6,10 Z M6,16 C4.9,16 4,16.9 4,18 C4,19.1 4.9,20 6,20 C7.1,20 8,19.1 8,18 C8,16.9 7.1,16 6,16 L6,16 Z"
    transform="translate(5, 1)"
  />
</svg>
`;

exports[`placeholder renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <rect
    height="24"
    rx="2"
    width="24"
    x="0"
    y="0"
  />
</svg>
`;

exports[`poll renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <path
    d="M19,3 L5,3 C3.9,3 3,3.9 3,5 L3,19 C3,20.1 3.9,21 5,21 L19,21 C20.1,21 21,20.1 21,19 L21,5 C21,3.9 20.1,3 19,3 Z M8,17 C7.45,17 7,16.55 7,16 L7,11 C7,10.45 7.45,10 8,10 C8.55,10 9,10.45 9,11 L9,16 C9,16.55 8.55,17 8,17 Z M12,17 C11.45,17 11,16.55 11,16 L11,8 C11,7.45 11.45,7 12,7 C12.55,7 13,7.45 13,8 L13,16 C13,16.55 12.55,17 12,17 Z M16,17 C15.45,17 15,16.55 15,16 L15,14 C15,13.45 15.45,13 16,13 C16.55,13 17,13.45 17,14 L17,16 C17,16.55 16.55,17 16,17 Z"
  />
</svg>
`;

exports[`search renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <path
    d="M16.6675008,14.6675008 L15.6141675,14.6675008 L15.2408341,14.3075008 C16.8408341,12.4408341 17.6675008,9.89416747 17.2141675,7.18750079 C16.5875008,3.48083412 13.4941675,0.52083412 9.76083413,0.0675007867 C4.12083412,-0.625832547 -0.625832547,4.12083412 0.0675007867,9.76083413 C0.52083412,13.4941675 3.48083412,16.5875008 7.18750079,17.2141675 C9.89416747,17.6675008 12.4408341,16.8408341 14.3075008,15.2408341 L14.6675008,15.6141675 L14.6675008,16.6675008 L20.3341675,22.3341675 C20.8808341,22.8808341 21.7741675,22.8808341 22.3208341,22.3341675 C22.8675008,21.7875008 22.8675008,20.8941675 22.3208341,20.3475008 L16.6675008,14.6675008 Z M8.66750079,14.6675008 C5.34750079,14.6675008 2.66750079,11.9875008 2.66750079,8.66750079 C2.66750079,5.34750079 5.34750079,2.66750079 8.66750079,2.66750079 C11.9875008,2.66750079 14.6675008,5.34750079 14.6675008,8.66750079 C14.6675008,11.9875008 11.9875008,14.6675008 8.66750079,14.6675008 Z"
  />
</svg>
`;

exports[`round_unchecked renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <path
    d="M12,21 C7.0275,21 3,16.9725 3,12 C3,7.0275 7.0275,3 12,3 C16.9725,3 21,7.0275 21,12 C21,16.9725 16.9725,21 12,21 Z M12,19 C15.8675,19 19,15.8675 19,12 C19,8.1325 15.8675,5 12,5 C8.1325,5 5,8.1325 5,12 C5,15.8675 8.1325,19 12,19 Z"
  />
</svg>
`;

exports[`uncheckbox renders correctly 1`] = `
.c0 {
  height: 24px;
  fill: #50bbff;
  width: 24px;
}

<svg
  className="c0"
  color="light"
  size={3}
  viewBox="0 0 24 24"
>
  <path
    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
  />
</svg>
`;
